### DISCUSSION TITLE
Trouble getting parcel source maps

### DISCUSSION THREAD

**Discussion by joshpetit** [parcel-bundler/parcel#6371] [Q&A]
I'm trying to get sourcemaps to work in parcel for a typescript project, but the generated source maps all lead to compiled (but still readable) files, this happens for both chrome and firefox. Looking at it, a `_parcel_source_root` is served which contains all the "slightly compiled" tsx/ts files. Has anyone experienced this? I've tried a variety of parcel versions (next, all the v2 betas and a few nightly ones also).

EDIT:
After further review, it seems that parcel is serving the wrong map file. There are 6 mapping files, and I renamed one of them to be the one that is served and everything seems to work! Is there some way to make it always use the correct source mappings everytime?

**Comment by DeMoorJasper** [comment]
This is pretty strange can you create a minimal reproduction for this?

**Reply by joshpetit** [reply]
I would but to be frank there are so many moving pieces. What I can say however is that we're using typescript and parcel serves the files with the command `parcel src/index.html -p 3000`. Sticking with the version on the npm page (2.0.0-beta.2) When just running that single command the source maps aren't even generated at all. This is the tsconfig:
```
{
  "compilerOptions": {
    "target": "esnext",
    "module": "esnext",
    "jsx": "preserve",
    "lib": ["dom", "es2017"],
    "moduleResolution": "node",
    "allowJs": false,
    "noEmit": true,
    "strict": true,
    "allowSyntheticDefaultImports": true,
    "skipLibCheck": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "removeComments": false,
    "preserveConstEnums": true,
    "sourceMap": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noImplicitAny": false,
    "types": ["jest"],
    "baseUrl": "./"
  },
  "exclude": ["node_modules"],
  "include": ["**/*.ts", "**/*.tsx", "**/*.svg"]
}
```
Sorry! If you can see a problem in this I'd appreciate it, but if not that's alright, thanks! Let me know if there are any other configurations that would be helpful.
